File: decimalSort/patrick.js
// Sort nums using only the fractional portion of each number.
// So under this ordering 50.40 is bigger than 100.09

function generateRandomDecimalNumbers() {
  var numbers = [];
  for (let i = 0; i < 100; i++) {
    const integerPart = Math.floor(Math.random() * 1000);
    const fractionalPart = Math.floor(Math.random() * 1000) / 1000;
    numbers.push(integerPart + fractionalPart);
  }
  return numbers;
}

const unsortedNumbers = generateRandomDecimalNumbers();

const sortedNumbers = unsortedNumbers.slice().sort((a, b) => {
  const fractionalPartA = a % 1;
  const fractionalPartB = b % 1;
  return fractionalPartA >= fractionalPartB ? 1 : -1;
});

console.log(sortedNumbers);

File: decimalSort/patrick.swift
#!/usr/bin/env swift
import Foundation

/* Problem
 * Sort nums using only the fractional portion of each number.
 * So under this ordering 50.40 is bigger than 100.09
*/

func generateRandomNumbers() -> [Float] {
    var numbers: [Float] = []
    for _ in 1...100 {
        let integerPart = Float(arc4random() % 1000);
        let fractionalPart = Float(arc4random() % 1000) / 1000.0
        numbers.append(integerPart + fractionalPart);
    }
    return numbers;
}

let sortedNumbers = generateRandomNumbers().sorted(by: { (a, b) -> Bool in
    let fractionalPartA = a.truncatingRemainder(dividingBy: 1);
    let fractionalPartB = b.truncatingRemainder(dividingBy: 1);
    return fractionalPartA <= fractionalPartB;
})

print(sortedNumbers)
